### DISCUSSION TITLE
Installing with alternate Python version

### DISCUSSION THREAD

**Discussion by AndroxxTraxxon** [StackStorm/st2#5964] [Q&A]
I'm attempting to install Stackstorm using Python3.11 as its runtime version in CentOS/Rocky. However, all the install scripts (default appear to install Python using Yum's python3-devel package. 

Does anyone know where I might alter the installation process to switch the python version? At the moment, it seems like the easiest way to do this might just be to build my own version of the `python3-devel` package with the python version I want so that the `yum` dependencies still resolve in the expected way.

I'm aware that there will be python package version issues after making this change, but I'm trying to take this one step at a time.

Additionally, I'm aware that there is a configuration option that lets you manage the python version for packs, which almost gets to what I want, but fails any time you try to run a Sensor with the following error (running the example Sensor on the "making a pack" tutorial):
```
Error processing line 1 of /opt/stackstorm/virtualenvs/hello_st2/lib/python3.11/site-packages/_virtualenv.pth:

Fatal Python error: init_import_site: Failed to import the site module
Python runtime state: initialized
Traceback (most recent call last):
  File "<frozen site>", line 186, in addpackage
  File "<string>", line 1, in <module>
  File "/opt/stackstorm/virtualenvs/hello_st2/lib/python3.11/site-packages/_virtualenv.py", line 41, in <module>
    from importlib.abc import MetaPathFinder
  File "/opt/stackstorm/st2/lib/python3.6/importlib/__init__.py", line 51, in <module>
    _w_long = _bootstrap_external._w_long
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'importlib._bootstrap_external' has no attribute '_w_long'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 982, in exec_module
  File "<frozen site>", line 616, in <module>
  File "<frozen site>", line 599, in main
  File "<frozen site>", line 531, in venv
  File "<frozen site>", line 384, in addsitepackages
  File "<frozen site>", line 226, in addsitedir
  File "<frozen site>", line 196, in addpackage
  File "/usr/local/lib/python3.11/traceback.py", line 5, in <module>
    import linecache
  File "/opt/stackstorm/st2/lib/python3.6/linecache.py", line 11, in <module>
    import tokenize
  File "/opt/stackstorm/st2/lib/python3.6/tokenize.py", line 33, in <module>
    import re
  File "/opt/stackstorm/st2/lib/python3.6/re.py", line 123, in <module>
    import sre_compile
  File "/opt/stackstorm/st2/lib/python3.6/sre_compile.py", line 17, in <module>
    assert _sre.MAGIC == MAGIC, "SRE module mismatch"
           ^^^^^^^^^^^^^^^^^^^
AssertionError: SRE module mismatch
```

Basically, I gather from this that Python3.6, which is being run by the st2sensorcontainer, is attempting to evaluate the python3.11 virtualenv, and in doing so is running into basic python builtin lib conflicts.

in light of this, I'd really like to attempt running with Python3.11, or find a way to alleviate this cross-version environment evaluation
